b2/b2.py

```diff
@@ -467,18 +467,21 @@ def ls(self, folder_to_list='', show_versions=False, max_entries=None, recursive
         # then we keep calling list_file_names until we get all of the
         # names in this "folder".
         current_dir = None
-        if show_versions:
-            api_name = 'b2_list_file_versions'
-        else:
-            api_name = 'b2_list_file_names'
-        url = url_for_api(self.api.account_info, api_name)
         start_file_name = prefix
         start_file_id = None
         while True:
             params = {'bucketId': self.id_, 'startFileName': start_file_name}
             if start_file_id is not None:
                 params['startFileId'] = start_file_id
-            response = post_json(url, params, auth_token)
+            raw_api = self.api.raw_api
+            api_url = self.api.account_info.get_api_url()
```

Comment:
after this change, `api_url` and `auth_token` are refreshed on every iteration of the loop. This will affect performance negatively.